equinix/fabric_connection_schema.go

```diff
@@ -0,0 +1,840 @@
+		},
+		"vlan_s_tag": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "Vlan Provider Tag information, vlanSTag value specified for QINQ connections",
+		},
+		"vlan_c_tag": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "Vlan Customer Tag information, vlanCTag value specified for QINQ connections",
+		},
+		"vni": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "vni",
+		},
+	}
+}
+
+var createAccessPointVirtualDeviceRes = &schema.Resource{
+	Schema: createAccessPointVirtualDeviceSch(),
+}
+
+func createAccessPointVirtualDeviceSch() map[string]*schema.Schema {
+
+	return map[string]*schema.Schema{
+		"href": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Unique Resource Identifier",
+		},
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Equinix-assigned Virtual Device identifier",
+		},
+		"type": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Virtual Device type",
+		},
+	}
+}
+
+func createAccessPointInterface() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Equinix-assigned interface identifier",
+		},
+		"id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "id",
+		},
+		"type": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Interface type",
+		},
+	}
+}
+
+func createFabricConnectionRoutingProtocol() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"type": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Routing Protocol type",
+		},
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Routing protocol instance identifier",
+		},
+		"state": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Routing protocol instance state",
+		},
+	}
+}
+
+var createPortRes = &schema.Resource{
+	Schema: createPortSch(),
+}
+
+func createPortSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Port information",
+		},
+		"href": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Unique Resource Identifier",
+		},
+		"name": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Port name",
+		},
+		"redundancy": {
+			Type:        schema.TypeSet,
+			Optional:    true,
+			Description: "Redundancy Information",
+			Elem: &schema.Resource{
+				Schema: createPortRedundancySch(),
+			},
+		},
+	}
+}
+
+var createConnectionSideAccessPointRes = &schema.Resource{
+	Schema: createConnectionSideAccessPointSch(),
+}
+
+func createConnectionSideAccessPointSch() map[string]*schema.Schema {
```

Comment:
There are a lot of [`Optional`](https://www.terraform.io/plugin/sdkv2/schemas/schema-behaviors#optional) fields here. Just checking that none of them are also [`Computed`](https://www.terraform.io/plugin/sdkv2/schemas/schema-behaviors#computed).
If they are computed and the API returns a default value for these optional fields, Terraform will want to reset the value back to empty unless the field is defined as `Computed`.

Reply:
These all are customer provided values, however; they are conditional, based on type of connection customer is creating. I have general concern on the comment "Terraform will reset to empty".  I will catchup with you offline.